v2/devdoc/logging.md

```diff
@@ -0,0 +1,339 @@
+    
+    // local context chaining another local context
+    LOG_CONTEXT_LOCAL_DEFINE(local_context_2, &local_context, LOG_CONTEXT_STRING_PROPERTY(other_knights_that_say, "%s", "Moo!"));
+    
+    LOGGER_LOG(LOG_LEVEL_VERBOSE, &local_context_2, "log with nee and moo");
+
+    log_from_a_function(&local_context_2);
+    
+    LOG_CONTEXT_DESTROY(context_2);
+}
+```
+
+The above example produces:
+
+```
+LOG_LEVEL_CRITICAL Time:Mon Aug 15 11:16:09 2022 File:G:\w\c-logging\v2\tests\logger_int\main.c:89 Func:main unnamed={ property_name=a  }  some critical error with context
+LOG_LEVEL_ERROR Time:Mon Aug 15 11:16:09 2022 File:G:\w\c-logging\v2\tests\logger_int\main.c:102 Func:main context_1={ name=haga  }  log error with context 1, value of some_var = 42
+LOG_LEVEL_WARNING Time:Mon Aug 15 11:16:09 2022 File:G:\w\c-logging\v2\tests\logger_int\main.c:103 Func:main context_2={ context_1={ name=haga  } last name=uaga age=42  }  log warning with context 2, some other string value is mumu
+LOG_LEVEL_ERROR Time:Mon Aug 15 11:16:09 2022 File:G:\w\c-logging\v2\tests\logger_int\main.c:16 Func:log_from_a_function context_2={ context_1={ name=haga  } last name=uaga age=42  }  log from a function!
+LOG_LEVEL_INFO Time:Mon Aug 15 11:16:09 2022 File:G:\w\c-logging\v2\tests\logger_int\main.c:112 Func:main context_name={ context_2={ context_1={ name=haga  } last name=uaga age=42  } the_knights_that_say=Nee!  }  log info with local context
+LOG_LEVEL_VERBOSE Time:Mon Aug 15 11:16:09 2022 File:G:\w\c-logging\v2\tests\logger_int\main.c:117 Func:main unnamed={ unnamed={ context_2={ context_1={ name=haga  } last name=uaga age=42  } the_knights_that_say=Nee!  } other_knights_that_say=Moo!  }  log with nee and moo
+LOG_LEVEL_ERROR Time:Mon Aug 15 11:16:09 2022 File:G:\w\c-logging\v2\tests\logger_int\main.c:16 Func:log_from_a_function unnamed={ unnamed={ context_2={ context_1={ name=haga  } last name=uaga age=42  } the_knights_that_say=Nee!  } other_knights_that_say=Moo!  }  log from a function!
+```
+
+## Log levels
+
+The following log levels shall be exposed by the library:
+
+- `CRITICAL`
+
+- `WARNING`
+
+- `ERROR`
+
+- `INFO`
+
+- `VERBOSE`
+
+## Sinks
+
+Multiple sinks shall be supported. Setting up globally which sinks to be used shall be supported (so that they do not have to be specified every time a `c_logging_log` call is made).
+
+Defining the sinks to use shall be done by defining a NULL terminated array, where each entry shall be a pointer to a structure that represents the sink interface.
+
+Example:
+
+```c
+const LOG_SINK* log_sinks[] = { &console_log_sink, &etw_log_sink, NULL };
+```
+
+### ETW sink
+
+An `ETW` sink shall be supported so that events are emitted via TraceLogging.
+
+For the `ETW` sink there should be a posibility of specifying the provider GUID to use.
+
+### console sink
+
+A console sink shall be supported so that events are printed to the console using `printf`.
```

Comment:
"... with support for multithreading so that the output is not intermingled".

Reply:
Added